packages/eslint-plugin/src/utils/ast-utils.ts

```
@@ -0,0 +1,485 @@
+          prop.key.expressions.length === 0
+        ) {
+          const value = prop.key.quasis[0]?.value.cooked
+          if (value) {
+            properties.push(value)
+          }
+        }
+        // Handle computed property keys with template literals: { [`name`]: ... }
+        else if (
+          prop.computed &&
+          prop.key.type === 'TemplateLiteral' &&
+          prop.key.expressions.length === 0
```

Biome (2.1.2)

[error] This branch can never execute. Its condition is a duplicate or covered by previous conditions in the if-else-if chain.



(lint/suspicious/noDuplicateElseIf)